service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/spring/KeycloakServiceImpl.java

```diff
@@ -403,6 +407,44 @@ public String seperator() {
 
 	private static final ObjectWriter WRITER = MAPPER.writer();
 
+	@Override
+	public int addClientScope(String clientScope) {
+		if (!isInitialized()) {
+			throw new RuntimeException("addClientScope(clientScope=" + clientScope + ") keycloak not initialized");
+		}
+
+		int updated = 0;
+		List<ClientScopeRepresentation> defaultClientScopes = realm.getDefaultDefaultClientScopes();
+		ClientScopeRepresentation clientScopeRepresentation = defaultClientScopes.stream().filter(s -> s.getName().equals(clientScope)).findAny().orElseThrow(() ->
+				new RuntimeException("client scope=" + clientScope + " doesn't exist"));
+		ClientsResource clients = realm.clients();
+		Collection<ClientRepresentation> clientsAll = clients.findAll(true);
+		Set<ClientRepresentation> clientIds = getRelevantClientIds(clientsAll);
+		for (ClientRepresentation clientRepresentation : clientIds) {
+			List<String> clientScopes = clientRepresentation.getDefaultClientScopes();
```

Comment:
can the list be `null`?

Reply:
I don't think so, because keycloak sets a few builtin default scopes